base/loading.jl

```diff
@@ -1146,9 +1146,13 @@ function create_expr_cache(input::String, output::String, concrete_deps::typeof(
                        --eval 'eval(Meta.parse(read(stdin,String)))'`, stderr=stderr),
               "w", stdout)
     # write data over stdin to avoid the (unlikely) case of exceeding max command line size
+    # pass PRECOMPILING_INTERACTIVELY state recursively
     write(io.in, """
-        Base.include_package_for_output($(repr(abspath(input))), $(repr(depot_path)), $(repr(dl_load_path)),
+        begin
+            Base.PRECOMPILING_INTERACTIVELY[] = $(PRECOMPILING_INTERACTIVELY[])
+            Base.include_package_for_output($(repr(abspath(input))), $(repr(depot_path)), $(repr(dl_load_path)),
             $(repr(load_path)), $deps, $(repr(uuid_tuple)), $(repr(source_path(nothing))))
+        end
```

Comment:
Can this be moved into `include_package_for_output`? It looks like that the motivation for introducing `include_package_for_output` was that it's a better style to put code in a function rather than in a function #36028. There was some precompilation speedup with the function approach.

Reply:
Makes sense. I've switched it over

Reply:
thanks btw